frontend/src/components/AddDatatsetServiceAreaSelector/AddDatasetServiceAreaSelector.tsx

```diff
@@ -0,0 +1,44 @@
+import { MenuItem, SelectField } from 'material-ui'
+import * as React from 'react'
+import { DatasetCountySelection } from '../../constants/datatypes'
+import { State } from '../../constants/states'
+import { Store, withStore } from '../../services/store'
+import { StateSelector } from '../Selectors/StateSelector'
+
+export let AddDatasetServiceAreaSelector = withStore(
+    'selectedState',
+    'addDatasetCountySelection'
+)(({ store }) =>
+    <div>
+        <StateSelector
+            onChange={state => onStateChange(state, store)}
+            value={store.get('selectedState')}
+        />
+        <CountySelector
+            onChange={value => store.set('addDatasetCountySelection')(value)}
+            value={store.get('addDatasetCountySelection')}
+        />
+    </div>
+)
+
+function onStateChange(state: State, store: Store) {
+    store.set('selectedState')(state)
+    store.set('uploadedServiceAreasFilename')(null)
```

Comment:
seems like this should maybe just go in the effects instead of having this separate function. we already have some things subscribed on state change. thoughts?